README.md

```diff
@@ -1,2 +1,94 @@
+- [Animation](#animation)  
+- [Selectors](#selectors)    
+- [Scoping](#scoping)  
+- [Grouping attributes within selectors](#grouping-attributes-within-selectors)  
+- [Code re-use](#code-re-use)  
+
+[JavaScript](#javascript) 
+- [State management](#state-management)  
+- [Component based architecture](#component-based-architecture)  
+- [Mitigate side effects](#mitigate-side-effects)   
+
+# Summary
+Goal of this document is for front-end developers to look up what would be the preferred solution for a given problem.  
+It is a collection of problems and potential solutions.  
+This document should also serve as a good way to navigate trough these solutions.  
+
+The information in this repo should allow developer to:
+
+- Find within reasonable time if this repo contains solution to the problem
+- Judge if the information is applicable for the encountered problem
+- Understand if/why the solution is preferred over other solutions
+- Implement the solution described in this repository
+- Judge if the information is outdated
+- See where the information is coming from
+
+#  What to Contribute
+Developers may contribute to this document via pull request.  
+Always be critical when to add more content.
+Searching in this document can also cost time.  
+When the answer to your question was found by a 1 minute google search it may not be a good problem to add here.  
+For problems and solutions to add to this repository at least one of the following is true:
+
+- Took considerable amount of time to look up through other channels (google, stack-overflow, other repo's)
+- Looking it up took more knowledge about the problem then would be considered normal for a junior developer.
+- It is difficult to select the right answer from a pile of not so good answers that pop up through google or stack-overflow
+- Our answer to the problem might be different from mainstream, because of our specific technology, domain, architecture, app, customer etcetera.
+
+# When to contribute
+Whenever you **implemented** OR **reviewed** a feature that:  
+- Is not yet in this repo  
+- Meets the requirements of [What to contribute](#what-to-contribute)
+
+
+# Contribute Format
+
+Please have a look at the first article is this repository:  
+[scale + wrap label and input depending on width](./code/wrap_label_input/README.md)  
+Also add the link under the relevant header below.  
+In the [Table of content](#table-of-content) add a "--> contains content" behind the link to mark that it has at least one article.  
+Please make sure all links are working so users can navigate fast.  
+If you disagree with any format/structure of this repository please let me know (Tom).  
+
+# Content
+For now I (Tom) focus on CSS, because that is where most chaos is.  
+I also find it hardest for CSS to find good answers online.  
+Feel free to add good JavaScript or HTML answers (or mixes of those).  
+# HTML
+# CSS
+## Which entities / Units ?
+### Pixels / REM / EM / % / VH 
```

Comment:
`VH or VW`
to make it apparent that it's also possible for width. ;)

Reply:
done